_conferences/__main__.py

```diff
@@ -1,117 +1,94 @@
-                "dates": dates_match[1],
-                "type": type_match[1],
-                "location": location_match[1],
-                "summary": summary_match[1],
-                "speaking": speaking_match[1] if speaking_match else "",
-            }
-            return conference
-    return None
-
-
-def normalize_url(url_match: str = None):
-    valid_url = None
-    # Ensure the url field is not blank and the url matches the regex
-    if url_match is not None and url_match.strip() != "":
-        # Parse the url and see if a scheme (`https`) is included in it
-        # If not, then prepend `https` to the url from the issue body
-        # This guards against the website thinking the passed in url is another page on https://blackpythondevs.com/
+    This guards against the website thinking the passed in url is another page on https://blackpythondevs.com/
+    """
+    if url_match:
         parsed_url = urlparse(url_match)
+
         if "http" not in parsed_url.scheme.casefold():
-            valid_url = f"https://{url_match}"
+            return f"https://{url_match}"
         else:
-            valid_url = url_match
-    return valid_url
+            return url_match
 
 
 def write_conferences_to_file(confs: list[dict]):
     # Write the conferences to the _data/conferences.yml file
-    with conferences_path.open("w") as f:
-        yaml.dump(confs, f)
-
-
-if __name__ == "__main__":
-    conferences = []
-
-    # Create Github client object
-    gh_client = create_github_client()
-
-    # Get open issues from repo
-    open_issues: PaginatedList[Issue] = get_open_issues(gh_client)
-
-    # Parse each conference issue so long as it has the "conference" label
-    for issue in open_issues:
-        if "conference" in [label.name for label in issue.labels]:
-            parsed_conf = parse_conference_details(issue_body=issue.body)
-            if parsed_conf:
-                conferences.append(parsed_conf)
-
-    write_conferences_to_file(conferences)
+    conferences_path.write_text(json.dumps(confs))
+
+
+def __to_conference_date(conference_date: str) -> datetime.date:
+    return datetime.date.fromisoformat(conference_date)
+
+
+def parse_conference(issue: gh_issues.Issue) -> dict[str, str | None]:
+    """convert an issue to a dictionary of parsed data"""
+
+    KEYS = [
+        "conference_name",
+        "url",
+        "conference_start_date",
+        "conference_end_date",
+        "conference_type",
+        "conference_location",
+        "summary",
+        "speaking",
+    ]
+
+    _issue = {k: getattr(issue, k, None) for k in KEYS}
+    _issue["url"] = normalize_url(_issue.get("url", None))
+    return _issue
+
+
+def _validate_issue(issue: gh_issues.Issue, date_to_check: str):
+    """Validate an issue based on its `date_to_check`"""
+    if not (valid_date := getattr(issue, date_to_check, False)):
+        return False
+    else:
+        return __to_conference_date(valid_date) >= datetime.date.today()

```

Comment:
Something else that I think would make sense to check is if `end_date >= start_date`. Take this for example:

```
today_date = 10/05/2024
start_date = 10/15/2024
end_date   = 10/10/2024
```

This scenario will always clear the `validate_issue` check because the conference end date is after today's date. But it still shouldn't be considered a valid conference since the end date is before the start date. Let me know if I'm overthinking this one or not

Reply:
I think maybe you're overthinking this as this validation is purely to filter out the entry from the generated json list.

These conference entries should still be validated by someone on the triage team and if there is an issue in entry, it is the triage members responsibility to detect that.